libraries/HTTPClient/src/HTTPClient.cpp

```diff
@@ -619,10 +622,21 @@ int HTTPClient::sendRequest(const char * type, uint8_t * payload, size_t size)
 
         // send Payload if needed
         if(payload && size > 0) {
-            if(_client->write(&payload[0], size) != size) {
-                return returnError(HTTPC_ERROR_SEND_PAYLOAD_FAILED);
+			
+			// this will fail for big payload
+//             if(_client->write(&payload[0], size) != size) {
+//                 return returnError(HTTPC_ERROR_SEND_PAYLOAD_FAILED);
+//             }
+			
+			// send the payload per 1000 bytes
+			log_d("Sending %u byte ...\n", size);
+            size_t sent_bytes = 0;
+			for (size_t id_data = 0; id_data < size; id_data+=1000)
```

Comment:
instead of a for loop using fixed sizes for chunks, please consider a loop similar to below (note this is not tested):
```
while(sent_bytes < size) {
  size_t sent = _client->write(&payload[sent_bytes], std::min(size - sent_bytes, _chunkSize));
  if (sent < 0) {
    log_e("Failed to send chunk!");
    break;
  }
  sent_bytes += sent;
}
```
Note that there is *ZERO* guarantee that the write call will send the full block size you request it to send thus you must check that it actually sent data and only then move the data starting point.  The check for negative sent byte count is to check for sending errors.

Reply:
@atanisoft Fixed. Please kindly check it. Thank you.